stl/inc/sstream

```diff
@@ -100,6 +150,70 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to a string directly, because
+    // the buffer may already be full, and the terminating char is not '\0'.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result(_Al);
+        if (!(_Mystate & _Constant) && _Mysb::pptr()) { // writable
+            const auto _Base   = _Mysb::pbase();
+            const auto _Ptr    = (_STD max)(_Mysb::pptr(), _Seekhigh);
+            const auto _Eptr   = _Mysb::epptr();
+            const auto _Mysize = static_cast<_Mysize_type>(_Ptr - _Base);
+            if (_Mysize > 0 && !_Traits::eq(*_Prev_iter(_Ptr), _Elem())) {
+                // the buffer is full and terminating char is not '\0'
+                _Result.assign(_Base, _Mysize);
+            } else {
+                _Result._Move_assign_from_buffer(_Base, _Mysize, static_cast<_Mysize_type>(_Eptr - _Base));
+                _Mystate &= ~_Allocated;
+            }
+        } else if (!(_Mystate & _Noread) && _Mysb::gptr()) { // readable
+            const auto _Base   = _Mysb::eback();
+            const auto _Eptr   = _Mysb::egptr();
+            const auto _Mysize = static_cast<_Mysize_type>(_Eptr - _Base);
+            if (_Mysize > 0 && !_Traits::eq(*_Prev_iter(_Eptr), _Elem())) {
+                // the buffer is full and terminating char is not '\0'
+                _Result.assign(_Base, _Mysize);
+            } else {
+                _Result._Move_assign_from_buffer(_Base, _Mysize, _Mysize);
+                _Mystate &= ~_Allocated;
+            }
+        }
+        _Tidy();
+        return _Result;
+    }
+
+    _NODISCARD _Mystr_view view() const noexcept {
+        const auto _Pair = _View_pair();
+        if (_Pair.first) {
+            return _Mystr_view{_Pair.first, _Pair.second};
+        } else {
+            return {};
+        }
+    }
+
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    void str(const basic_string<_Elem, _Traits, _Alloc2>& _Newstr) {
+        _Tidy();
+        _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
+    }
+
+    void str(_Mystr&& _Newstr) {
+        _Tidy();
+        _Init_inplace(_STD move(_Newstr), _Mystate);

```

Comment:
This is not correctly handling POCMA

Reply:
Yes, we have been talking about this. I'll handle it.